File: backend/api-gateway-server/src/main/java/com/myportfolio_apigateway/authentication/dto/UserDto.java
package com.myportfolio_apigateway.authentication.dto;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDto {

    private int userId;
    private String userName;
    private String surName;
    private String emailId;
    private String password;
    private String address;
    private String mobileNo;
    private int age;
    private Gender gender;
    private double height;
    private double weight;
    private double bmiIndex;
    private BMICategory bmiCategory;

    private String profilePicUrl;
    private Date createdAt;
    private Date updatedAt;

}

File: frontend/src/app/models/user.dto.ts
export enum Gender {
  MALE = 'MALE',
  FEMALE = 'FEMALE',
  PREFER_NOT_TO_SAY = 'PREFER_NOT_TO_SAY'
}

export enum BMICategory {
  UNDERWEIGHT = 'UNDERWEIGHT',
  NORMAL = 'NORMAL',
  OVERWEIGHT = 'OVERWEIGHT',
  OBESITY = 'OBESITY'
}

export interface UserDto {
  userId: number;
  userName: string;
  surName: string;
  emailId: string;
  password: string;
  address: string | null;
  mobileNo: string | null;
  age: number;
  gender: string;
  height: number;
  weight: number;
  bmiIndex: number;
  bmiCategory: string;
  profilePicUrl: string | null;
  createdAt: string;
  updatedAt: string;
} 